lib/pages/settings/service/admin_provider.dart

```diff
@@ -0,0 +1,123 @@
+import 'package:acs_upb_mobile/authentication/model/user.dart';
+import 'package:acs_upb_mobile/authentication/service/auth_provider.dart';
+import 'package:acs_upb_mobile/generated/l10n.dart';
+import 'package:acs_upb_mobile/pages/settings/model/request.dart';
+import 'package:acs_upb_mobile/widgets/toast.dart';
+import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+
+extension RequestExtension on Request {
+  static Request fromSnap(DocumentSnapshot snap) {
+    final data = snap.data();
+    return Request(
+      userId: data['addedBy'],
+      requestBody: data['requestBody'],
+      processed: data['done'],
+      type: RequestType.permissions,
+      dateSubmitted: data['dateSubmitted'],
+      formId: snap.id,
+    );
+  }
+
+  static String getFormId(DocumentSnapshot snap) {
+    return snap.id;
+  }
+}
+
+class AdminProvider with ChangeNotifier {
+  final FirebaseFirestore _db = FirebaseFirestore.instance;
+
+  Future<List<String>> fetchAllRequests() async {
+    try {
+      final QuerySnapshot qSnapshot = await _db
+          .collection('forms')
+          .orderBy('dateSubmitted', descending: false)
+          .get();
+      return qSnapshot.docs.map(RequestExtension.getFormId).toList();
+    } catch (e) {
+      print(e);
+      AppToast.show(S.current.errorSomethingWentWrong);
+      return null;
+    }
+  }
+
+  Future<List<String>> fetchUnprocessedRequests() async {
+    try {
+      final QuerySnapshot qSnapshot = await _db
+          .collection('forms')
+          .where('done', isEqualTo: false)
+          .orderBy('dateSubmitted', descending: false)
+          .get();
+      return qSnapshot.docs.map(RequestExtension.getFormId).toList();
+    } catch (e) {
+      print(e);
+      AppToast.show(S.current.errorSomethingWentWrong);
```

Comment:
This is a super bad practice that we have in many places in code, which makes it very hard to debug when we have a million generic "something went wrong" messages. We eventually need to fix this approach, but for now, at least show a more informative error message (e.g. "Could not load requests").